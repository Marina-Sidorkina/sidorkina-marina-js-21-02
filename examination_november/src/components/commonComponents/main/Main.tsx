import React from 'react';
import './Main.scss';

const Main = (props: { children: React.ReactChild | React.ReactFragment | null | undefined; }) => (
  <main className="main">
    <div className="main__container">
      { props.children }
    </div>
  </main>
);

export default Main;
