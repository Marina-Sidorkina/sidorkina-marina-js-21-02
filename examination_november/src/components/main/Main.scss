.main {
  position: relative;
  display: flex;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
  max-width: 2000px;
  min-width: 320px;
  height: 100%;

  &__container {
    width: 100%;
    max-width: 1200px;
    min-width: 320px;
    min-height: 100%;
    padding: 15px;
  }
}
