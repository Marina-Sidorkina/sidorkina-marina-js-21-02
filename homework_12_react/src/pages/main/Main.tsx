import React from "react";
import Promo from "../../components/promo/Promo";
import {promo} from "../../mocks/promo";
import Categories from "../../components/categories/Categories";
import {categories} from "../../mocks/categories";
import Favourite from "../../components/favourite/Favourite";
import {pianos} from "../../mocks/pianos";

class Main extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Promo title={ promo.title } message={ promo.message }/>
        <Categories categories={ categories }/>
        <Favourite favourites={ pianos }/>
      </React.Fragment>
    )
  }
}

export default Main;
