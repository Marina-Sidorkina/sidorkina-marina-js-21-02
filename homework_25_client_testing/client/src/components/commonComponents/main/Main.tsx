import React from 'react';
import styles from './Main.module.scss';

const Main = (props: { children: React.ReactChild | React.ReactFragment | null | undefined; }) => (
  <main className={styles.main}>
    <div className={styles.container}>
      { props.children }
    </div>
  </main>
);

export default Main;
