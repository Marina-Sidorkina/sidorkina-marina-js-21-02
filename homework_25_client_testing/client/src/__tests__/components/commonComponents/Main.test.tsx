import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
import Main from '../../../components/commonComponents/main/Main';

configure({ adapter: new Adapter() });

describe('Main component test', () => {
  test('should render Main', () => {
    const wrapper = shallow(<Main>{}</Main>);
    expect(wrapper.length).toEqual(1);
  });

  test('should render children', () => {
    const wrapper = shallow(<Main><div className="children">Children</div></Main>);
    expect(wrapper.find('div.children').text()).toBe('Children');
  });

  test('should render Main', () => {
    const wrapper = shallow(<Main><div className="children">Children</div></Main>);
    expect(wrapper).toMatchSnapshot();
  });
});
