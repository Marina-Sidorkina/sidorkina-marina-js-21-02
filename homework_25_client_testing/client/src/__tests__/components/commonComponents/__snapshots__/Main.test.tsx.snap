// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component test should render Main 1`] = `
<main
  className="main"
>
  <div
    className="container"
  >
    <div
      className="children"
    >
      Children
    </div>
  </div>
</main>
`;
